samples/solutions/provision_symmetric_key_failover_device_delete.py

```diff
@@ -0,0 +1,374 @@
+                        await asyncio.sleep(SLEEP_TIME_BETWEEN_CHECKING_REGISTRATION)
+                        self.iothub_assignment_fail_event.set()
+                        self.iothub_assignment_sucess_event.clear()
+                except Exception as e:
+                    self.log_error_and_print(
+                        "Registration process failed because of error {}".format(get_type_name(e))
+                    )
+                    raise Exception(
+                        "Caught an unrecoverable error that needs to be "
+                        "fixed from user end while registering. Will exit application..."
+                    )
+                finally:
+                    await provisioning_client.shutdown()
+
+            if self.exit_app_event.is_set():
+                return
+
+    async def wait_for_connect_and_send_telemetry(self):
+        id = 1
+        while True:
+            if not self.iothub_client:
+                # Time to check if device has been provisioned
+                self.log_info_and_print(
+                    "IoTHub client is still nonexistent for telemetry. "
+                    "Will check after {} secs...".format(SLEEP_TIME_BETWEEN_CHECKING_REGISTRATION)
+                )
+                await asyncio.sleep(SLEEP_TIME_BETWEEN_CHECKING_REGISTRATION)
+            elif not self.iothub_client.connected:
+                self.log_info_and_print("IoTHub client is existent. But waiting for connection ...")
+                await self.connected_event.wait()
+            else:
+                self.log_info_and_print("sending message with id {}....".format(id))
+                await self.iothub_client.send_message("message number {}".format(id))
+                id += 1
+                self.log_info_and_print("sent message.....")
+                self.log_info_and_print("sleeping for {} secs...".format(TELEMETRY_INTERVAL))
+                await asyncio.sleep(TELEMETRY_INTERVAL)
+            if self.exit_app_event.is_set():
+                return
+
+    async def if_disconnected_then_connect_with_retry(self):
+        while True:
+            self.log_info_and_print("Entry for retry after disconnection")
+            done, pending = await asyncio.wait(
+                [
+                    self.disconnected_event.wait(),
+                    self.exit_app_event.wait(),
+                ],
+                return_when=asyncio.FIRST_COMPLETED,
+            )
+            self.log_info_and_print("Exit for retry after disconnection")
+            await asyncio.gather(*done)
+            [x.cancel() for x in pending]
+            if self.exit_app_event.is_set():
+                self.log_info_and_print("Exiting while connected")
+                return
+            if not self.iothub_client or self.iothub_assignment_fail_event.is_set():
+                self.log_info_and_print(
+                    "IoTHub client is still nonexistent or re-provisioning is needed for establishing connection."
```

Comment:
maybe better to say:
IoTHub client is invalid or unassigned, re-provisioning is required to proceed.